### PR TITLE
Revert iovr=4 (exponential cloud overlap method) in RRTMG

### PR DESCRIPTION
**Pull request by mzhangw**
# PR Checklist

- [x] Ths PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR. Please consult the ufs-weather-model [wiki](https://github.com/ufs-community/ufs-weather-model/wiki/Making-code-changes-in-the-UFS-weather-model-and-its-subcomponents) if you are unsure how to do this.

- [x] This PR has been tested using a branch which is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR

- [x] An Issue describing the work contained in this PR has been created either in the subcomponent(s) or in the ufs-weather-model. The Issue should be created in the repository that is most relevant to the changes in contained in the PR. The Issue and the dependent sub-component PR 
are specified below.

- [ ] Results for one or more of the regression tests change and the reasons for the changes are understood and explained below.

- [ ] New or updated input data is required by this PR. If checked, please work with the code managers to update input data sets on all platforms.

## Description

This effort is to revert/cleanup iovr=4 in RRTMG per developer's request. The current exponential cloud overlapping method used in RRTMG represents the method used in the pre-2018 operational HWRF model. The associated ccpp-physics PR is https://github.com/NCAR/ccpp-physics/pull/780

The exponential cloud overlap method used  in pre-2018 operational HWRF (iovr=4) is reverted per developer's request. It will affect all RTs that use iovr=4. 

### Issue(s) addressed

Issue 1 in https://github.com/NCAR/ccpp-physics/issues/748

## Testing

This PR has been tested on Hera.

Full regression tests will be run on all tier-1 platforms at the time of commit.

- [x] hera.intel
- [x] hera.gnu
- [x] orion.intel
- [skip - maintenance Dec 27-31] cheyenne.intel 
- [skip - maintenance Dec 27-31] cheyenne.gnu
- [x] gaea.intel 
- [x] jet.intel
- [x] wcoss_cray
- [x] wcoss_dell_p3
- [n/a] opnReqTest for newly added/changed feature
- [x] CI - https://github.com/ufs-community/ufs-weather-model/pull/963/commits/8dbecba3cd7c39db09ba70c03c486136293facc6

## Dependencies

- waiting on https://github.com/NCAR/ccpp-physics/pull/780
- waiting on https://github.com/NOAA-EMC/fv3atm/pull/445

### CODE REVIEW

**Comment by climbfuji** [comment]
@mzhangw Please use the template to provide the necessary information instead of the current 2-liner at the top of the PR description. Also, is there an fv3atm PR?

**Comment by mzhangw** [comment]
It is to disable iovr=4 option in RTs using RRTMG in pre-2018 HWRF favor.
We need to keep the metadata info in fv3atm since RRTMGP still uses EXP and
EXP-random cloud overlap method. So no PR from fv3atm is needed.

**Comment by climbfuji** [comment]
> It is to disable iovr=4 option in RTs using RRTMG in pre-2018 HWRF favor. We need to keep the metadata info in fv3atm since RRTMGP still uses EXP and EXP-random cloud overlap method. So no PR from fv3atm is needed.
> […](#)

Sure, but let me repeat to please fill in the PR description template that the UFS code managers have provided. Also, you still need to create an fv3atm PR to handle the submodule pointer update for ccpp-physics. Please see https://github.com/ufs-community/ufs-weather-model/wiki/Making-code-changes-in-the-UFS-weather-model-and-its-subcomponents and https://github.com/ufs-community/ufs-weather-model/wiki/UFS-weather-model-code-commit-tutorial-recordings. Thanks!

**Comment by grantfirl** [comment]
@mzhangw This one also needs the .gitmodules and fv3atm submodule pointer updated

File: tests/rt.conf [hunk 1]
```diff
@@ -91,8 +91,8 @@ RUN     | control_ras
 RUN     | control_thompson                                                                                                        | - wcoss_cray                            | fv3 |
 RUN     | control_thompson_no_aero                                                                                                | - wcoss_cray                            | fv3 |
 
-RUN     | fv3_HAFS_v0_hwrf_thompson                                                                                               |                                         | fv3 |
-RUN     | fv3_esg_HAFS_v0_hwrf_thompson                                                                                           |                                         | fv3 |
+#RUN     | fv3_HAFS_v0_hwrf_thompson                                                                                               |                                         | fv3 |
```
Comment (climbfuji) [inline tests/rt.conf]: Please remove these lines entirely

File: tests/rt.conf [hunk 2]
```diff
@@ -138,9 +138,9 @@ RUN     | rap_flake_debug
 RUN     | rap_noah_sfcdiff_cires_ugwp_debug                                                                                       |                                         | fv3 |
 RUN     | rrfs_v1beta_debug                                                                                                       |                                         | fv3 |
 
-COMPILE | -DAPP=ATM -DCCPP_SUITES=FV3_HAFS_v0_hwrf_thompson -DDEBUG=ON                                                            |                                         | fv3 |
-RUN     | fv3_HAFS_v0_hwrf_thompson_debug                                                                                         |                                         | fv3 |
-RUN     | fv3_esg_HAFS_v0_hwrf_thompson_debug                                                                                     | -cheyenne.intel                         | fv3 |
+#COMPILE | -DAPP=ATM -DCCPP_SUITES=FV3_HAFS_v0_hwrf_thompson -DDEBUG=ON                                                            |                                         | fv3 |
```
Comment (climbfuji) [inline tests/rt.conf]: Please remove these lines entirely

File: tests/rt_gnu.conf [hunk 3]
```diff
@@ -2,7 +2,7 @@
 # CCPP PROD tests                                                                                                                                                #
 ##################################################################################################################################################################
 
-COMPILE | -DAPP=ATM -DCCPP_SUITES=FV3_GFS_v16,FV3_GFS_v16_flake,FV3_GFS_v16_RRTMGP,FV3_GFS_v16_thompson,FV3_GFS_v16_ras,FV3_HAFS_v0_hwrf_thompson,FV3_GFS_v16_nsstNoahmpUGWPv1 | | fv3 |
+COMPILE | -DAPP=ATM -DCCPP_SUITES=FV3_GFS_v16,FV3_GFS_v16_flake,FV3_GFS_v16_RRTMGP,FV3_GFS_v16_thompson,FV3_GFS_v16_ras,FV3_GFS_v16_nsstNoahmpUGWPv1 | | fv3 |
```
Comment (climbfuji) [inline tests/rt_gnu.conf:5]: We should also delete the suite definition file(s) in your fv3atm PR

File: tests/rt_gnu.conf [hunk 4]
```diff
@@ -15,8 +15,8 @@ RUN     | control_thompson_no_aero
 RUN     | control_ras                                                                                                             |                | fv3         |
 RUN     | control_p7                                                                                                              |                | fv3         |
 
-RUN     | fv3_HAFS_v0_hwrf_thompson                                                                                               |                | fv3         |
-RUN     | fv3_esg_HAFS_v0_hwrf_thompson                                                                                           |                | fv3         |
+#RUN     | fv3_HAFS_v0_hwrf_thompson                                                                                               |                | fv3         |
```
Comment (climbfuji) [inline tests/rt_gnu.conf]: Please remove these lines entirely

File: tests/rt_gnu.conf [hunk 5]
```diff
@@ -58,9 +58,9 @@ RUN     | control_ras_debug
 RUN     | control_stochy_debug                                                                                                    |                | fv3         |
 RUN     | control_debug_p7                                                                                                        |                | fv3         |
 
-COMPILE | -DAPP=ATM -DCCPP_SUITES=FV3_HAFS_v0_hwrf_thompson -DDEBUG=ON                                                            |                | fv3         |
-RUN     | fv3_HAFS_v0_hwrf_thompson_debug                                                                                         |                | fv3         |
-RUN     | fv3_esg_HAFS_v0_hwrf_thompson_debug                                                                                     |                | fv3         |
+#COMPILE | -DAPP=ATM -DCCPP_SUITES=FV3_HAFS_v0_hwrf_thompson -DDEBUG=ON                                                            |                | fv3         |
```
Comment (climbfuji) [inline tests/rt_gnu.conf]: Please remove these lines entirely

File: tests/tests/fv3_HAFS_v0_hwrf_thompson [hunk 6]
```diff
@@ -1,112 +0,0 @@
...
-                   RESTART/phy_data.tile5.nc \
-                   RESTART/phy_data.tile6.nc"
-
-
-export_fv3
-export NODES=$(expr $TASKS / $TPN + 1)
-
-DT_ATMOS="600"
-
-export FV3_RUN=c96_HAFS_v0_hwrf_run.IN
```
Comment (climbfuji) [inline tests/tests/fv3_HAFS_v0_hwrf_thompson:80]: We should also remove in`tests/fv3_conf`:
```
c96_HAFS_v0_hwrf_run.IN
esg_HAFS_v0_hwrf_run.IN
esg_HAFS_v0_hwrf-model_configure.IN
```

**Comment by climbfuji** [comment]
One last one to delete, please (in fv3_conf): esg_HAFS_v0_hwrf-model_configure.IN

**Comment by climbfuji** [comment]
@DusanJovic-NOAA @BinLiu-NOAA Changes look good now. I added the run-ci, hera-intel-RT and hera-gnu-RT labels for now. Let's wait until they come back. There should be no baseline changes.

I think Cheyenne is under maintenance the entire week, we'll have to skip the platform with its two compilers.

Unfortunately the pull request template got overwritten, I'll put back the necessary items.

**Comment by climbfuji** [comment]
@DusanJovic-NOAA @BrianCurtis-NOAA @MinsukJi-NOAA @mzhangw The jobs passed on Hera with Intel and GNU, but the problem is that emc.nemspara can't push to @mzhangw's branch. Because the bug in auto-rt still hasn't been fixed, the log files and run directries got deleted.

Is there anything we can do so that we can still use auto-rt? I don't want to run manually on all platforms. Thanks ...

**Comment by DusanJovic-NOAA** [comment]
> @DusanJovic-NOAA @BrianCurtis-NOAA @MinsukJi-NOAA @mzhangw The jobs passed on Hera with Intel and GNU, but the problem is that emc.nemspara can't push to @mzhangw's branch. Because the bug in auto-rt still hasn't been fixed, the log files and run directries got deleted.
> 
> Is there anything we can do so that we can still use auto-rt? I don't want to run manually on all platforms. Thanks ...

I am not familiar with Auto-RT scripts, but this is very annoying. If 'git push' is not successful, script should not delete anything.

**Comment by climbfuji** [comment]
> > @DusanJovic-NOAA @BrianCurtis-NOAA @MinsukJi-NOAA @mzhangw The jobs passed on Hera with Intel and GNU, but the problem is that emc.nemspara can't push to @mzhangw's branch. Because the bug in auto-rt still hasn't been fixed, the log files and run directries got deleted.
> > Is there anything we can do so that we can still use auto-rt? I don't want to run manually on all platforms. Thanks ...
> 
> I am not familiar with Auto-RT scripts, but this is very annoying. If 'git push' is not successful, script should not delete anything.

Agreed. I just created #983 which is identical to #963 but from my user. auto-rt should be able to push there. Once testing is completed, @mzhangw can pull the regression test logs from my branch and this PR will be closed.

**Comment by mzhangw** [comment]
I will pull from here when it is ready. Thanks!

**Comment by climbfuji** [comment]
@mzhangw @DusanJovic-NOAA I was able to push my branch with the regression test logs to @mzhangw's branch. All regression test logs are there. I will now merge ccpp-physics.

**Review by DusanJovic-NOAA — APPROVED** [review]

**Comment by climbfuji** [comment]
Hash for fv3atm is correct (42dd808) please check again and merge if ok. Thanks!

**Pull request merged**